### ISSUE TITLE
Incorrect form validation when creating an auth source

### ISSUE THREAD

**Issue by kimburgess** [PlaceOS/backoffice#168]
When adding a SAML2 IdP, the form validation is currently enforcing entry on optional fields. Specifically: `Name Identifier Format`. Internal docs currently indicate this may be optionally specified. This appears to be reinforced by there being a [default value in the model](https://github.com/PlaceOS/models/blob/d6147c04fe2043b44e902e1fed7279ffe3b05cc7/src/placeos-models/saml_authentication.cr#L22).

Defaults also exist for other attributes—`attribute_statements`, `request_attributes`, `issuer`—however these are not presented to the user.

Similar behaviour may exist for other providers or general entity creation.

![FE364208-BE89-42F8-A50E-AFE5CF7C7FA3](https://user-images.githubusercontent.com/843652/112907801-c9343900-9131-11eb-8576-002653c31199.jpeg)

**Comment by MrYuion** [comment]
Hmm... I think I was just following what Steve had given to me. 
I've updated `Name Identifier Format` and `Request Attributes` to not be required.

**Comment by kimburgess** [comment]
Yep, for usability there's probably value in exposing what the defaults are as these attributes _do_ need a value and presence of this is validated on the backend. They just have a default during initial creation if unspecified at that time.

This is a more general problem rather that just for this modal and likely a little more complex as AFAIK this information is not available from the API.

CC @stakach
